lib/trino-filesystem-s3/src/test/java/io/trino/filesystem/s3/TestS3FileSystemAwsS3.java

```diff
@@ -68,4 +78,32 @@ private static String environmentVariable(String name)
     {
         return requireNonNull(System.getenv(name), "Environment variable not set: " + name);
     }
+
+    @Test
+    void testS3FileIteratorFileEntryTags()
+            throws IOException
+    {
+        try (S3Client s3Client = createS3Client()) {
```

Comment:
add a test for the restored object as well?

Reply:
I believe testing for restored glacier object may be a little difficult since glacier restores are extremely slow. Testing glacier objects is more intuitive as we can mock an s3 bucket with a glacier object; however, in the putobject request there is no way we can manipulate the restore status